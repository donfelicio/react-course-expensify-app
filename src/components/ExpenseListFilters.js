import React from "react";
import { connect } from "react-redux";
import {
	setTextFilter,
	sortByAmount,
	sortByDate,
	setStartDate,
	setEndDate
} from "../actions/filters";
import { DateRangePicker } from "react-dates";

export class ExpenseListFilters extends React.Component {
	state = {
		calendarFocused: null
	};

	onDatesChange = ({ startDate, endDate }) => {
		this.props.setStartDate(startDate);
		this.props.setEndDate(endDate);
	};

	onFocusChange = calendarFocused => {
		this.setState(() => ({ calendarFocused }));
	};

	onTextChange = e => {
		this.props.setTextFilter(e.target.value);
	};

	onSortChange = e => {
		switch (e.target.value) {
			case "amount":
				return this.props.sortByAmount();
			default:
				return this.props.sortByDate();
		}
	};

	render() {
		return (
			<div className="content-container">
				<div className="input-group">
					<div className="input-group__item">
						{" "}
						<input
							type="text"
							placeholder="Search expenses"
							className="text-input"
							value={this.props.filters.text}
							onChange={this.onTextChange}
						/>
					</div>
					<div className="input-group__item">
						{" "}
						<select
							value={this.props.filters.sortBy}
							className="select"
							onChange={this.onSortChange}
						>
							<option value="date">Date</option>
							<option value="amount">Amount</option>
						</select>
					</div>
					<div className="input-group__item">
						{" "}
						<DateRangePicker
							startDate={this.props.filters.startDate}
							startDateId={
								this.props.filters.startDate
									? this.props.filters.startDate.toString()
									: ""
							}
							endDate={this.props.filters.endDate}
							endDateId={
								this.props.filters.endDate
									? this.props.filters.endDate.toString()
									: ""
							}
							onDatesChange={this.onDatesChange}
							focusedInput={this.state.calendarFocused}
							onFocusChange={this.onFocusChange}
							numberOfMonths={1}
							isOutsideRange={() => {
								false;
							}}
							showClearDates={true}
						/>
					</div>
				</div>
			</div>
		);
	}
}

const mapDispatchToProps = dispatch => ({
	setTextFilter: text => dispatch(setTextFilter(text)),
	sortByDate: () => dispatch(sortByDate()),
	sortByAmount: () => dispatch(sortByAmount()),
	setStartDate: startDate => dispatch(setStartDate(startDate)),
	setEndDate: endDate => dispatch(setEndDate(endDate))
});

const mapStateToProps = state => ({
	filters: state.filters
});

export default connect(
	mapStateToProps,
	mapDispatchToProps
)(ExpenseListFilters);
