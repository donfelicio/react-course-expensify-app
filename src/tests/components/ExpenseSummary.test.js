import React from "react";
import { shallow } from "enzyme";
import expenses from "../fixtures/expenses";
import { ExpenseSummary } from "../../components/ExpenseSummary";

test("Should render the ExpenseSummary component", () => {
	const expensesTotal = 100;
	const wrapper = shallow(
		<ExpenseSummary
			expenseCount={expenses.length}
			expenseTotal={expensesTotal}
		/>
	);
	expect(wrapper).toMatchSnapshot();
});

test("Should render the filtered ExpenseSummary component", () => {
	const expensesTotal = 195;
	const wrapper = shallow(
		<ExpenseSummary expenseCount={1} expenseTotal={expensesTotal} />
	);
	expect(wrapper).toMatchSnapshot();
});

test("Should render the ExpenseSummary with an empty input", () => {
	const wrapper = shallow(
		<ExpenseSummary expenseCount={0} expenseTotal={0} />
	);
	expect(wrapper).toMatchSnapshot();
});
