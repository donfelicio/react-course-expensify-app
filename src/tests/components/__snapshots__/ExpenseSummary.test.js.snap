// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the ExpenseSummary component 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        3
      </span>
       
      expenses
       totalling
       
      <span>
        $1.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`Should render the ExpenseSummary with an empty input 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        0
      </span>
       
      expenses
       totalling
       
      <span>
        $0.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`Should render the filtered ExpenseSummary component 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling
       
      <span>
        $1.95
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add expense
      </Link>
    </div>
  </div>
</div>
`;
